block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -155,15 +154,17 @@ impl BlockWitness {
         let advice_inputs = {
             let mut merkle_store = MerkleStore::default();
             merkle_store
-                .add_merkle_paths(self.updated_accounts.into_iter().map(
+                .add_merkle_paths(self.updated_accounts.iter().map(
                     |(
                         account_id,
                         AccountUpdate {
                             initial_state_hash,
                             final_state_hash: _,
                             proof,
                         },
-                    )| { (u64::from(account_id), initial_state_hash, proof) },
+                    )| {
+                        (u64::from(*account_id), *initial_state_hash, proof.clone())
```

Comment:
This `proof.clone()` could be expensive - any reason why we change `self` -> `&self`?

Reply:
humm, I don't recall why I changed into a non-consuming version. Let me fix that